assets/js/admin/woocommerce_admin.js

```diff
@@ -296,4 +296,14 @@ jQuery( function ( $ ) {
 			$( this ).replaceWith( '<iframe width="560" height="315" src="https://www.youtube-nocookie.com/embed/' + video_id + '" frameborder="0" allowfullscreen></iframe>' );
 		} );
 	});
+
+	// Validate email colors settings
+	$( '#woocommerce_email_base_color[type=text], #woocommerce_email_background_color[type=text], #woocommerce_email_body_background_color[type=text], #woocommerce_email_text_color[type=text]' ).on( 'change', function() {
+		var hexcolor = jQuery(this).val();
+		if (! hexcolor.match(/^\#([a-fA-F0-9]{6}|[a-fA-F0-9]{3})$/)) {
```

Comment:
Check your spaces, coding standards say there needs to be spaces before and after brackets.